iroha/src/asset.rs

```diff
@@ -6,101 +6,103 @@ use parity_scale_codec::{Decode, Encode};
 pub struct Asset {
     /// identifier of asset, formatted as asset_name#domain_id
     pub id: Id,
-    pub amount: u128,
+    pub quantity: u128,
```

Comment:
Will rust always use u128 worth of memory for this, or is it smart enough to use less for lesser numbers? Very few use cases need this many digits, although some indeed do.

Reply:
Maybe we can use [BigInt](https://docs.rs/num-bigint/0.2.6/num_bigint/) here, which is essentially a vector of digits, so it will scale up and down. I think it might be better both in terms of storage and the fact that maybe some account would have more asset than the upper limit of u128.

Reply:
We would have to test the size implications.

Reply:
https://soramitsu.atlassian.net/browse/RH2-88?atlOrigin=eyJpIjoiMWY2NmQ2NDVhNjc3NGE5NWIyNGQ2MDY3NTIwOTQwOTciLCJwIjoiaiJ9